### DISCUSSION TITLE
Pass arguments across different tasks

### DISCUSSION THREAD

**Discussion by DmitryLukyanov** [cake-build/cake#3717] [Q&A] [answered]
Is there a way to pass arguments across different tasks?

For example I have task A that can be called before task B and C:
   
    Task("A")
      .Does(() =>
      {   });

    Task("B")
      .IsDependentOn("A")
      .Does(() =>
      {   };

    Task("C")
      .IsDependentOn("A")
      .Does(() =>
      {   };

I want to configure behavior inside task A based on where it's called from, something like:

    Task("A")
      .Does((arg) =>
      {
           if (arg == "Value1") { .. }
           else { .. }    
      });

    Task("B")
      .IsDependentOn("A", "Value1")
      .Does(() =>
      {   };

    Task("C")
      .IsDependentOn("A", "Value2")
      .Does(() =>
      {   };

**Comment by augustoproiete** [comment, marked as answer]
Yes, there's a couple of different ways to share data between tasks. See [Sharing Build State](https://cakebuild.net/docs/writing-builds/sharing-build-state) in the docs
